src/hotspot/share/oops/method.cpp

```diff
@@ -2254,10 +2254,12 @@ bool Method::is_method_id(jmethodID mid) {
 Method* Method::checked_resolve_jmethod_id(jmethodID mid) {
   if (mid == NULL) return NULL;
   Method* o = resolve_jmethod_id(mid);
-  if (o == NULL || o == JNIMethodBlock::_free_method || !((Metadata*)o)->is_method()) {
+  if (o == NULL || o == JNIMethodBlock::_free_method) {
```

Comment:
I need to think more about deleting the is_method() check.

Reply:
The is_method() check was from when JNIHandleBlocks contained both jmethodID's and jobjects.  It's a leftover from permgen elimination. I should have mentioned that in the PR comments.  Now the only things that can be put in the JNIMethodBlocks are Methods, which the compiler type-checks, so that's the only thing that can come out (unless corrupted).

Reply:
Thanks for the explanation.